src/test/java/co/rsk/federate/signing/hsm/advanceBlockchain/HSMBookKeepingClientProviderTest.java

```diff
@@ -0,0 +1,55 @@
+package co.rsk.federate.signing.hsm.advanceblockchain;
+
+import co.rsk.federate.signing.hsm.HSMClientException;
+import co.rsk.federate.signing.hsm.HSMUnsupportedVersionException;
+import co.rsk.federate.signing.hsm.client.HSMBookkeepingClient;
+import co.rsk.federate.signing.hsm.client.HSMClientProtocol;
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.Test;
+
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+
+/**
+ * Created by Kelvin Isievwore on 27/03/2023.
+ */
+public class HSMBookKeepingClientProviderTest {
+
+    private HSMBookKeepingClientProvider hsmBookKeepingClientProvider;
+    private HSMClientProtocol hsmClientProtocol;
```

Comment:
```suggestion
    private HSMClientProtocol hsmClientProtocol;
```

Also, correct this `HSM Client Protocol`, I believe it should be named using camel case no matter "HSM" is an acronym. In fact, you used camel case when naming the implementation `HsmBookkeepingClientImpl`. So, let's be consistent with the naming.

Reply:
I Don't Understand Your Suggestion.. Elaborate Further?

Reply:
Sure. And sorry the case I meant for class names is PascalCase, not camelCase, with the first letter of each separate word capitalized, no matter if is an acronym.

Take a look at `HsmBookkeepingClientImpl` class. You used PascalCase, but in other classes, you capitalized all the acronym`HSM`, for example for `HSMClientProtocol` class. So, my suggestion is to use only one style, this `Hsm` or this one `HSM`.

Take a look at this answer: https://stackoverflow.com/questions/2236807/java-naming-convention-with-acronyms

Reply:
`HSM` Is The Standard For HSM Related Classes, I'll Change That For `HsmBookkeepingClientImpl` But For Variable/Field Names, `hsmXXX` Is Fairly Okay. Thanks For The Update